src/prototype_creep_fight.js

```diff
@@ -342,128 +342,357 @@ Creep.prototype.fightRanged = function(target) {
 };
 
-Creep.prototype.siege = function() {
-  this.memory.hitsLost = this.memory.hitsLast - this.hits;
-  this.memory.hitsLast = this.hits;
+Creep.prototype.avoidEdge = function(range) {
+  let lowerRangeModifier = range - 1;
+  let higherRangeModifier = 50 - range;
+  if (this.pos.x <= lowerRangeModifier ||
+    this.pos.x >= higherRangeModifier ||
+    this.pos.y <= lowerRangeModifier ||
+    this.pos.y >= higherRangeModifier
+  ) {
+    this.moveTo(25, 25, {
+      reusePath: 0
+    });
+    return true;
+  }
+  return false;
+};
 
-  if (this.hits - this.memory.hitsLost < this.hits / 2) {
-    let exitNext = this.pos.findClosestByRange(FIND_EXIT);
-    this.moveTo(exitNext);
+Creep.prototype.fightSafeMode = function() {
+  if (this.room.controller.safeMode) {
+    let constructionSites = this.room.find(FIND_CONSTRUCTION_SITES);
+    this.moveTo(constructionSites[0], {
+      reusePath: 0,
+      ignoreCreeps: true
+    });
     return true;
   }
+  return false;
+};
 
-  if (!this.memory.notified) {
+Creep.prototype.notifyAttack = function() {
+  if (config.autoattack.notify && !this.memory.notified) {
     this.log('Attacking');
     Game.notify(Game.time + ' ' + this.room.name + ' Attacking');
     this.memory.notified = true;
   }
-  var tower = this.pos.findClosestByRange(FIND_HOSTILE_STRUCTURES, {
+};
+
+Creep.prototype.siege = function() {
+
+  this.notifyAttack();
+
+  if (this.fightSafeMode()) {
+    return true;
+  }
+
+  var spawn = this.pos.findClosestByRange(FIND_HOSTILE_STRUCTURES, {
     filter: function(object) {
-      if (object.structureType == STRUCTURE_TOWER) {
-        return true;
+      return ((object.structureType == STRUCTURE_SPAWN || object.structureType == STRUCTURE_TOWER) && !object.pos.lookFor(LOOK_STRUCTURES)[1]);
+    }
+  });
+  if (spawn === null) {
+    spawn = this.pos.findClosestByRange(FIND_HOSTILE_STRUCTURES, {
+      filter: function(object) {
+        return (object.structureType == STRUCTURE_SPAWN || object.structureType == STRUCTURE_TOWER);
       }
-      if (object.structureType == STRUCTURE_CONTROLLER) {
-        return true;
+    });
+  }
+
+  if (spawn === null) {
+    let structures = this.pos.findClosestByRange(FIND_HOSTILE_STRUCTURES, {
+      filter: function(object) {
+        return object.structureType != STRUCTURE_CONTROLLER;
       }
+    });
+    if (structures === null) {
+      this.say('YouGotPwnd', true);
+      var constructionSites = this.pos.findClosestByRange(FIND_CONSTRUCTION_SITES);
+      this.moveTo(constructionSites, {
+        reusePath: 0,
+        ignoreCreeps: true
+      });
       return true;
     }
+    this.moveTo(structures, {
+      reusePath: 0,
+      ignoreCreeps: true
+    });
+    this.dismantle(structures);
+    return true;
+  }
+
+  let search = PathFinder.search(
+    this.pos, {
+      pos: spawn.pos,
+      range: 1
+    }, {
+      maxRooms: 1
+    }
+  );
+
+  var healer = this.pos.findInRange(FIND_MY_CREEPS, 45, {
```

Comment:
Somehow comment why `45`. To exclude creeps on the other side of the room.


But this does only work for creeps on the other side of the room.